activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -182,6 +182,24 @@ def drop_current(environment = env)
         }
       end
 
+      def truncate_tables(configuration)
+        ActiveRecord::Base.establish_connection(configuration)
+        table_names = ActiveRecord::Base.connection.tables
+        internal_table_names = [
+          ActiveRecord::Base.schema_migrations_table_name,
+          ActiveRecord::Base.internal_metadata_table_name
```

Comment:
Seems like this might have a need for something similar to what we're doing here: https://github.com/rails/rails/pull/33985/files#diff-18d566ee525de376b1a3741881545631R147

Reply:
There is one more place that possible should be refactored
https://github.com/rails/rails/blob/84718df86097442f85999d6f2e6f6b8b59724c3f/activerecord/lib/active_record/schema_dumper.rb#L256. I'm not sure that we can use something like `ActiveRecord::Base.descendants.select { |model| !model._internal? }.map(&:table_name).compact` here (It ignores join tables, ...) since  It isn't the same like `ActiveRecord::Base.connection.tables`.